### PR TITLE
Handle missing attributes when patching a moves documents

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

P4-<TODO>

### What?

Adds support for patching a move without any attribute changes (i.e. only the documents)

### Why?

This is a genuine use case of this api.

### CODE REVIEW

File: spec/requests/api/v1/moves_controller_update_spec.rb [hunk 1]
```diff
@@ -162,6 +162,34 @@
               response_json.dig('data', 'relationships', 'documents', 'data').map { |document| document['id'] },
             ).to eq(after_documents.pluck(:id))
           end
+
+          context 'when there are no attributes in the patch' do
```
Comment (willfish) [inline spec/requests/api/v1/moves_controller_update_spec.rb:166]:
```suggestion
          context 'when there are no attributes in the params' do
```

File: spec/requests/api/v1/moves_controller_update_spec.rb [hunk 2]
```diff
@@ -162,6 +162,34 @@
...
+            it 'does not affect other relationships' do
+              expect { do_patch }.not_to change { move.reload.from_location }
+            end
+
+            it 'returns the updated documents in the response body' do
+              do_patch
+
+              expect(
+                response_json.dig('data', 'relationships', 'documents', 'data').map { |document| document['id'] },
+              ).to eq(after_documents.pluck(:id))
```
Comment (cwrw) [inline spec/requests/api/v1/moves_controller_update_spec.rb:190]: if order is not an issue here could we use `match_array` to avoid the risk of flakiness?

Comment (willfish) [inline spec/requests/api/v1/moves_controller_update_spec.rb:190]:
```suggestion
              ).to match_array(after_documents.pluck(:id))
```

File: spec/requests/api/v1/moves_controller_update_spec.rb [hunk 3]
```diff
@@ -162,6 +162,34 @@
...
+            let(:move_params) do
+              documents = after_documents.map { |d| { id: d.id, type: 'documents' } }
+              {
+                type: 'moves',
+                relationships: { documents: { data: documents } },
+              }
+            end
+
+            it 'updates the moves documents' do
+              expect(move.reload.documents).to eq(before_documents)
```
Comment (willfish) [inline spec/requests/api/v1/moves_controller_update_spec.rb:176]:
```suggestion
              expect(move.reload.documents).to match_array(before_documents)
```

File: spec/requests/api/v1/moves_controller_update_spec.rb [hunk 4]
```diff
@@ -162,6 +162,34 @@
...
+              do_patch
+              expect(move.reload.documents).to eq(after_documents)
```
Comment (willfish) [inline spec/requests/api/v1/moves_controller_update_spec.rb:178]:
```suggestion
              expect(move.reload.documents).to match_array(after_documents)
```

File: spec/requests/api/v1/moves_controller_update_spec.rb (hunk 2 above)
Comment (willfish) [reply spec/requests/api/v1/moves_controller_update_spec.rb]: So this is all very weird. It makes me think we're always relying on the natural ordering in the db which is a bit temperamental. We should fix that separately.

Comment (martyn-w) [reply spec/requests/api/v1/moves_controller_update_spec.rb]: i've fixed this in my (unrelated) pr - I'd just changed `eq` to `match_array` as we can't assume an order from the database

**Pull request merged**